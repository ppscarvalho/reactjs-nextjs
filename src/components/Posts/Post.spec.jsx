import { render, screen } from '@testing-library/react';
import { Posts } from '.';

const props = {
  posts: [
    {
      id: 1,
      title: 'Teste 1',
      body: 'body 1',
      cover: 'img/imagem1.png',
    },
    {
      id: 2,
      title: 'Teste 2',
      body: 'body 2',
      cover: 'img/imagem2.png',
    },
    {
      id: 3,
      title: 'Teste 3',
      body: 'body 3',
      cover: 'img/imagem3.png',
    },
  ],
};

describe('<Posts />', () => {
  it('should render posts', () => {
    render(<Posts {...props} />);
    expect(screen.getAllByRole('heading', { name: /Teste/i })).toHaveLength(3);
    expect(screen.getAllByRole('img', { name: /Teste/i })).toHaveLength(3);
    expect(screen.getAllByText(/body/i)).toHaveLength(3);
    expect(screen.getByRole('img', { name: /Teste 3/i })).toHaveAttribute('src', 'img/imagem3.png');
  });

  it('should not render posts', () => {
    render(<Posts />);
    expect(screen.queryAllByRole('heading', { name: /Teste/i })).toHaveLength(0);
  });

  it('should match snapshot', () => {
    const { container } = render(<Posts {...props} />);
    expect(container).toMatchSnapshot();
  });
});
