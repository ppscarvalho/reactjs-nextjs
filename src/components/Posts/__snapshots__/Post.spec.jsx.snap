// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Posts /> should match snapshot 1`] = `
<div>
  <div
    class="posts"
  >
    <div
      class="post"
    >
      <img
        alt="Teste 1"
        src="img/imagem1.png"
      />
      <div
        class="post-content"
      >
        <h1>
          Teste 1
           
          1
        </h1>
        <p>
          body 1
        </p>
        <p>
          Alterado
        </p>
      </div>
    </div>
    <div
      class="post"
    >
      <img
        alt="Teste 2"
        src="img/imagem2.png"
      />
      <div
        class="post-content"
      >
        <h1>
          Teste 2
           
          2
        </h1>
        <p>
          body 2
        </p>
        <p>
          Alterado
        </p>
      </div>
    </div>
    <div
      class="post"
    >
      <img
        alt="Teste 3"
        src="img/imagem3.png"
      />
      <div
        class="post-content"
      >
        <h1>
          Teste 3
           
          3
        </h1>
        <p>
          body 3
        </p>
        <p>
          Alterado
        </p>
      </div>
    </div>
  </div>
</div>
`;
